pkg/webhook/ratelimit.go

```diff
@@ -0,0 +1,91 @@
+package webhook
+
+import (
+	"sync"
+	"time"
+)
+
+// RateLimiter implements a sliding window rate limiting algorithm
+type RateLimiter struct {
+	mu       sync.Mutex
+	clients  map[string][]int64
+	lastSeen map[string]time.Time
+	window   time.Duration
+	allowed  int
+	done     chan bool
+}
```

Comment:
Can you add docs to this new type, especially 2 maps fields, preferrably with some sample data to show their usage?